src/common/imageio_libraw.c

```diff
@@ -53,32 +53,142 @@ static gboolean _supported_image(const gchar *filename)
 }
 
 
-dt_imageio_retval_t dt_imageio_open_libraw(dt_image_t *img, const char *filename, dt_mipmap_buffer_t *mbuf)
+
+int dt_libraw_lookup_makermodel(const char *maker, const char *model,
+                                   char *mk, int mk_len, char *md, int md_len,
+                                   char *al, int al_len)
 {
-  int err = DT_IMAGEIO_FILE_CORRUPTED;
+  int got_it_done = FALSE;
 
-  if(!_supported_image(filename)) return DT_IMAGEIO_FILE_CORRUPTED;
+  if(g_str_equal(maker, "Canon"))
+  {
+    g_strlcpy(mk, "Canon", mk_len);
 
-  if(!img->exif_inited) (void)dt_exif_read(img, filename);
+    if(g_str_equal(model, "Canon EOS RP"))
+    {
+      g_strlcpy(md, "EOS RP", md_len);
```

Comment:
Yes and no, because it would not handle different brand names for the same model. rawspeed provides us a clean_make and clean_model name. We need our own logic for libraw.

Reply:
at least for Canon EOS M50,  Canon EOS M50m2,  Canon EOS M200,  Canon EOS R,  Canon EOS RP,  Canon EOS 250D,  Canon EOS 90D,  Canon EOS 850D, Canon EOS M6 Mark II,  Canon PowerShot G7 X Mark III,  Canon PowerShot G5 X Mark II,  Canon EOS-1D X Mark III,  Canon EOS R5,  Canon EOS R6 removing prefix "maker" works

Reply:
> I don't know why, but exiv2 reports **Canon** for make and **Canon EOS RP** for model.

This is actually the pristine metadata in the file, you'd get the same w/ exiftool...

Reply:
i tried with this and checked for KISS and REBEL SL3 - but that doesn't work for Powershot Sx70 HS:
```
  if(g_str_equal(maker, "Canon"))
  {
    g_strlcpy(mk, "Canon", mk_len);

    if(g_str_equal(model, "Canon EOS 250D") ||
    g_str_equal(model, "Canon EOS 850D") ||
    g_str_equal(model, "Canon EOS 90D") ||
    g_str_equal(model, "Canon EOS-1D X Mark III") ||
    g_str_equal(model, "Canon EOS M50") ||
    g_str_equal(model, "Canon EOS M50m2") ||
    g_str_equal(model, "Canon EOS M200") ||
    g_str_equal(model, "Canon EOS M6 Mark II") ||
    g_str_equal(model, "Canon EOS R") ||
    g_str_equal(model, "Canon EOS RP") ||
    g_str_equal(model, "Canon EOS R5") ||
    g_str_equal(model, "Canon EOS R6") ||
    g_str_equal(model, "Canon PowerShot G7 X Mark III") ||
    g_str_equal(model, "Canon PowerShot G5 X Mark II") ||
    g_str_equal(model, "Canon PowerShot SX70 HS"))
    {
      g_strlcpy(md, g_strjoinv("", g_strsplit(model, "Canon ", -1)), md_len);
      g_strlcpy(al, g_strjoinv("", g_strsplit(model, "Canon ", -1)), al_len);
    }
    
    if(g_str_equal(model, "Canon EOS KISS M"))
    {
      g_strlcpy(md, "EOS M50", md_len);
      g_strlcpy(al, "EOS KISS M", al_len);
    }    
  
    if(g_str_equal(model, "Canon EOS KISS M2"))
    {
      g_strlcpy(md, "EOS M50m2", md_len);
      g_strlcpy(al, "EOS KISS M2", al_len);
    }    
  
  if(g_str_equal(model, "Canon EOS KISS X10"))
    {
      g_strlcpy(md, "EOS 250D", md_len);
      g_strlcpy(al, "EOS KISS X10", al_len);
    }    

  if(g_str_equal(model, "Canon EOS KISS X10i"))
    {
      g_strlcpy(md, "EOS 850D", md_len);
      g_strlcpy(al, "EOS KISS X10i", al_len);
    }    

  if( g_str_equal(model, "Canon EOS Rebel SL3"))
    {
      g_strlcpy(md, "EOS 250D", md_len);
      g_strlcpy(al, "EOS Rebel SL3", al_len);
    }    

  if(g_str_equal(model, "Canon EOS Rebel T8i"))
    {
      g_strlcpy(md, "EOS 850D", md_len);
      g_strlcpy(al, "EOS Rebel T8i", al_len);
    }    

    got_it_done = TRUE;
  }

```